src/main/java/frc/robot/subsystems/coral/CoralScoring.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems.coral;
+
+import org.littletonrobotics.junction.Logger;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/*
+ * Coral Scoring Subsystems
+ */
+/*
+ *
+ */
+public class CoralScoring extends SubsystemBase {
+    private CoralScoringIO io;
+    private CoralScoringInputsAutoLogged coralScoringAutoLogged =
+        new CoralScoringInputsAutoLogged();
+
+    public CoralScoring(CoralScoringIO io) {
+        this.io = io;
+        io.updateInputs(coralScoringAutoLogged);
+    }
+
+    @Override
+    public void periodic() {
+        io.updateInputs(coralScoringAutoLogged);
+        Logger.processInputs("Coral Scoring", coralScoringAutoLogged);
+    }
+
+    public void setScoringMotor(double percentage) {
+        Logger.recordOutput("Scoring Percentage", percentage);
+        io.setCoralScoringMotorPercentage(percentage);
+    }
+
+    public boolean getScoringBeamBrakeStatus() {
+        return coralScoringAutoLogged.scoringBeamBrake;
+    }
+
+    /**
+     * Command that returns a command
+     */
+
+    private Command motorStartEndCommand(double scoringSpeed) {
+        return Commands.startEnd(() -> {
+            setScoringMotor(scoringSpeed);
+        }, () -> {
+            setScoringMotor(0);
+        }, this);
+    }
+
+    /**
+     * Runs Pre Scoring Motor
+     */
+    public Command runPreScoringMotor(double scoringSpeed) {
+        return motorStartEndCommand(scoringSpeed).until(() -> getScoringBeamBrakeStatus());
```

Comment:
I would add a timeout on this so it stops after some time like if the coral falls out or something, maybe like 10 seconds.  See https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Command.html#withTimeout(double)